File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)

        dp = [[-1 for _ in range(n)] for _ in range(n)]

        def check(i, j):
            if i > j:
                return True
            if i == j:
                return True
            if s[i] != s[j]:
                return False
            if dp[i][j] != -1:
                return dp[i][j]
            dp[i][j] = check(i + 1, j - 1)
            return dp[i][j]

        ans = start = 0
        for i in range(n):
            for j in range(i, n):
                if j - i + 1 > ans and check(i, j):
                    ans = j - i + 1
                    start = i
        return s[start : start + ans]

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cs
public class Solution {
    public string LongestPalindrome(string s) {
        int max = -1;
        var res= "" ;   
        for(int i =0 ; i < s.Length; i++)
        {
            for(int j = i ; j < s.Length; j++)
            {
                if(Ispalindromic(s,i,j))
                {
                    if(max < j - i + 1)
                    {
                        max = j - i + 1;
                        res = s.Substring(i, j-i+1); 
                    }
                }
            }
        }
        return res;
    }

    bool Ispalindromic(string s , int i ,int j)
    {
        if( i >= j )
        {
            return true;
        }

        if(s[i] != s[j]) return false;

        return Ispalindromic(s, i+1, j-1);
    }

}
